ddev/tests/cli/release/branch/test_create.py

```diff
@@ -30,9 +30,51 @@ def test_create_branch(ddev, mocker):
         mocker.call('checkout', 'master'),
         mocker.call('pull', 'origin', 'master'),
         mocker.call('checkout', '-B', '5.5.x'),
+        mocker.call('add', '.gitlab/build_agent.yaml'),
+        mocker.call('commit', '-m', 'Update build_agent.yaml to use agent branch: 5.5.x'),
         mocker.call('push', 'origin', '5.5.x'),
     ]
 
     assert create_label_mock.call_args_list == [
         mocker.call('backport/5.5.x', '5319e7'),
     ]
+
+
+def test_update_build_agent_yaml_success(mocker, tmp_path):
+    """Test successful update of build_agent.yaml"""
+    from ddev.cli.release.branch.create import update_build_agent_yaml
+    from ddev.utils.fs import Path
+
+    # Create a temporary build_agent.yaml file
+    build_agent_path = Path(tmp_path / '.gitlab' / 'build_agent.yaml')
+    build_agent_path.parent.ensure_dir_exists()
+
+    original_content = """---
+.build-agent-tpl:
+  variables:
+    RELEASE_VERSION_6: "nightly"
+    RELEASE_VERSION_7: "nightly-a7"
+  trigger:
+    project: DataDog/datadog-agent
+    branch: main
+    strategy: depend
+
+build-agent-auto:
+  extends: .build-agent-tpl
+  rules:
+    - if: $CI_COMMIT_BRANCH =~ /^7\\.\\d+\\.x$/
+      when: never
+"""
+
+    build_agent_path.write_text(original_content)
+
+    app_mock = mocker.MagicMock()
+
+    with Path(tmp_path).as_cwd():
+        update_build_agent_yaml(app_mock, '7.99.x')
+
+    updated_content = build_agent_path.read_text()
+
+    assert 'branch: 7.99.x' in updated_content
+    assert 'branch: main' not in updated_content
+    app_mock.display_success.assert_called_once_with('Updated build_agent.yaml file to use Agent branch: 7.99.x')
```

Comment:
nit: Since we already checked that the branch name has changed I would probably not assert this. Unless we want to define this print as a key feature we want to protect. The reason is that if tomorrow we want to say something different we are going to come and change the test.

Anyway, not a blocker.